setup.py

```diff
@@ -1,44 +1,45 @@
 from setuptools import find_packages, setup
 
-# Read the content of the README file	
-with open('README.md', encoding='utf-8') as f:	
-    long_description = f.read()	
+# Read the content of the README file
+with open("README.md", encoding="utf-8") as f:
+    long_description = f.read()
 
-# Read the content of the requirements.txt file	
-with open('requirements.txt', encoding='utf-8') as f:	
-    requirements = f.read().splitlines()	
+# Read the content of the requirements.txt file
+with open("requirements.txt", encoding="utf-8") as f:
+    requirements = f.read().splitlines()
 
-setup(	
-    name="dspy-ai",	
-    version="2.4.3",	
-    description="DSPy",	
-    long_description=long_description,	
-    long_description_content_type='text/markdown',	
-    url="https://github.com/stanfordnlp/dsp",	
-    author="Omar Khattab",	
-    author_email="okhattab@stanford.edu",	
-    license="MIT License",	
-    packages=find_packages(include=['dsp.*', 'dspy.*', 'dsp', 'dspy']),	
-    python_requires='>=3.9',	
-    install_requires=requirements,	
+setup(
+    name="dspy-ai",
+    version="2.4.3",
+    description="DSPy",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/stanfordnlp/dsp",
+    author="Omar Khattab",
+    author_email="okhattab@stanford.edu",
+    license="MIT License",
+    packages=find_packages(include=["dsp.*", "dspy.*", "dsp", "dspy"]),
+    python_requires=">=3.9",
+    install_requires=requirements,
     extras_require={
         "chromadb": ["chromadb~=0.4.14"],
         "qdrant": ["qdrant-client", "fastembed"],
         "marqo": ["marqo~=3.1.0"],
-        "mongodb": ["pymongo~=3.12.0"],			
-        "pinecone": ["pinecone-client~=2.2.4"],	
-        "weaviate": ["weaviate-client~=3.26.1"],	
+        "mongodb": ["pymongo~=3.12.0"],
+        "pinecone": ["pinecone-client~=2.2.4"],
+        "weaviate": ["weaviate-client~=3.26.1"],
         "faiss-cpu": ["sentence_transformers", "faiss-cpu"],
         "milvus": ["pymilvus~=2.3.7"],
         "google-vertex-ai": ["google-cloud-aiplatform==1.43.0"],
-    },	
-    classifiers=[	
-        "Development Status :: 3 - Alpha",	
-        "Intended Audience :: Science/Research",	
-        "License :: OSI Approved :: MIT License",	
-        "Operating System :: POSIX :: Linux",	
-        "Programming Language :: Python :: 3",	
-        "Programming Language :: Python :: 3.8",	
-        "Programming Language :: Python :: 3.9",	
-    ],	
-)	
+        "snowflake": ["snowflake-snowpark-python"],
```

Comment:
since this added as an extras (aka dspy-ai[snowflake]), can you add similar logic to how this handled in other RM imports (example with [Milvus](https://github.com/stanfordnlp/dspy/blob/d7dace6b51e1c0899fd798c575262310939f6deb/dspy/retrieve/milvus_rm.py#L14))  in [SnowflakeRM](https://github.com/stanfordnlp/dspy/pull/859/files#diff-d982e89c27806bdc8dca2b257dab30c39f85db9a9dbbbabdb28ae15ddd550cd8R15)

Reply:
and then also update the README to add this as a supported extras dependency - line starting with "For the optional ..."

Reply:
Done